core/rx-java-driver/src/main/java/org/hawkular/rx/cassandra/driver/RxSessionImpl.java

```diff
@@ -38,9 +43,14 @@
 public class RxSessionImpl implements RxSession {
 
     private Session session;
+    private int maximumInflightRequests;
 
     public RxSessionImpl(Session session) {
         this.session = session;
+        maximumInflightRequests = Math.min(
```

Comment:
The max number of inflight requests overall is num_hosts * max_connections_per_host * max_requests_per_connection